### PR TITLE
Enable/Disable user-functionality mgt rest API

### PR DESCRIPTION
**Pull request by piraveena**
Depends on this RP https://github.com/wso2/carbon-identity-framework/pull/3068. once it is merged, need to bump framework version

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                    statusChangeRequest.getFunctionalityLockReason());
+            return Response.ok().build();
+        } catch (UserFunctionalityManagementException e) {
+            if (e instanceof UserFunctionalityManagementClientException) {
+                throw handleUserFunctionalityMgtClientException(e);
+            }
+            log.error("Error occurred while calling user functionality management service.", e);
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_LOCK_THE_FUNCTIONALITY_FAILED);
+
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java:108]: remove redundant line

File: components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                        Constants.ErrorMessages.ERROR_CODE_USER_NOT_PERMITTED_TO_UNLOCK);
+            }
+        } catch (UserFunctionalityManagementException e) {
+            if (e instanceof UserFunctionalityManagementClientException) {
+                throw handleUserFunctionalityMgtClientException(e);
+            }
+            log.error("Error occurred while calling user functionality management service.", e);
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_LOCK_THE_FUNCTIONALITY_FAILED);
+
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java:222]: remove redundant line

File: components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                            .getFunctionalityLockReason());
+            return Response.ok().build();
+        } catch (UserFunctionalityManagementException e) {
+            if (e instanceof UserFunctionalityManagementClientException) {
+                throw handleUserFunctionalityMgtClientException(e);
+            }
+            log.error("Error occurred while calling user functionality management service.", e);
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_LOCK_THE_FUNCTIONALITY_FAILED);
+
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java:266]: remove redundant line

File: components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_GET_LOCK_STATUS_FAILED);
+        }
+    }
+
+    private APIError handleUserFunctionalityMgtClientException(UserFunctionalityManagementException e) {
+
+        if (StringUtils.equals(UserFunctionalityMgtConstants.ErrorMessages.USER_NOT_FOUND.getCode(),
+                e.getErrorCode())) {
+            log.debug("Cannot find the user corresponding to the userId.", e);
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java:298]: should this be a debug log or an error log? if it's a debug log please check if debug logs are enabled first.

Comment (sachiniWettasinghe) [reply components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java:298]: check other instances as well

Comment (piraveena) [reply components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java:298]: It should be debug log. If it is error logs, then console will be flooded withlogs when a invalid request is sent by the user. User may try multiple times with invalid request and logs will be  flooded

**Review by sachiniWettasinghe — APPROVED** [review]

**Pull request merged**